### PR TITLE
feat: [CDS-79225]: Add harness release entity to service configuration & outcome

### PR DESCRIPTION
**Pull request by abhinavsingh1196**
## Describe your changes
- Add release object to service YAML for K8s/Native Helm service types
- Add release object to service step outcome `<+service.release.name>`

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53028)

### CODE REVIEW

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/service/ServiceSpec.java [hunk 1]
```diff
@@ -31,4 +32,7 @@ public interface ServiceSpec {
   List<NGVariable> getVariables();
   List<ConfigFileWrapper> getConfigFiles();
   @JsonIgnore String getType();
+  default HarnessRelease getRelease() {
```
Comment (abosii-harness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/service/ServiceSpec.java:35]: I believe we shouldn't include release field in the other services spec

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Comment by abhinavsingh1196** [comment]
trigger utstrigger uts

**Comment by abhinavsingh1196** [comment]
trigger uts

**Review by abosii-harness — APPROVED** [review]

**Comment by abhinavsingh1196** [comment]
trigger uts

**Review by alebuh — DISMISSED** [review]

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/service/beans/NativeHelmServiceSpec.java` line 87 at r3](https://reviewable.io/reviews/harness/harness-core/53028#-Neg13Rs7O-XsH0KFyqW:-Neg13Rs7O-XsH0KFyqX:b-585dkz) ([raw file](https://github.com/harness/harness-core/blob/68f9c83ab7ddb0db11326854f7c8f491f749eda8/127-cd-nextgen-entities/src/main/java/io/harness/cdng/service/beans/NativeHelmServiceSpec.java#L87)):*
> ```Java
> 
>     children.add("manifestConfigurations", manifestConfigurations);
>     children.add("release", release);
> ```

Just fYI, this is not needed because `HarnessRelease` does not contain any reference to any Harness entity.

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 3 of 8 files at r1, 5 of 5 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53028)* status: 9 file(s) to review. 1. unresolved discussions. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepV3.java` line 768 at r3](https://reviewable.io/reviews/harness/harness-core/53028#-Neg-tzJ5YLdWMcoA9g8:-Neg-tzJ5YLdWMcoA9g9:blwas2p) ([raw file](https://github.com/harness/harness-core/blob/68f9c83ab7ddb0db11326854f7c8f491f749eda8/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepV3.java#L768)):*
> ```Java
>       HarnessRelease harnessRelease = harnessReleaseOptional.get();
>       stepOutcomes.add(StepResponse.StepOutcome.builder()
>                            .name(OutcomeExpressionConstants.RELEASE_OUTCOME)
> ```

we generally don't append outcomes by the word "outcome"
In this case, users will access this by `harnessReleaseOutcome.name`
IMO `harnessRelease.name` is less verbose

**Comment by abhinavsingh1196** [comment]
trigger uts

**Comment by abhinavsingh1196** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]

**Review by alebuh — APPROVED** [review]

**Comment by abhinavsingh1196** [comment]
trigger review

**Pull request merged**